src/main/internal-lifecycle.js

```diff
@@ -27,19 +27,33 @@ const cleanUpBeforeClosingSync = () => {
   rimraf.sync(path.resolve(LEDGER_CONFIG_DIRECTORY, "sqlite/*.log"));
 };
 
-const handleExit = (worker: ?cluster$Worker, code, signal) => {
-  const pid = String(worker);
-  console.log(`worker ${pid} died with error code ${code} and signal ${signal}`);
-  logger.warn(`Internal process ended with code ${code}`);
+const handleExit = (code, signal) => {
+  if (!internalProcess) {
+    console.log("something went very wrong in the process lifecycle management");
+    return;
+  }
+
+  if (code !== null) {
+    console.log(`internal process ${internalProcess.pid} gracefully exited with code ${code}`);
+    logger.warn(`Internal process ended with code ${code}`);
+  } else {
+    console.log(`internal process ${internalProcess.pid} got killed by signal ${signal}`);
+    logger.warn(`Internal process killed with signal ${signal}`);
+  }
   internalProcess = null;
 };
 
 const killInternalProcess = () => {
   if (internalProcess) {
-    logger.log("killing internal process...");
-    internalProcess.removeListener("exit", handleExit);
-    internalProcess.kill("SIGINT");
-    internalProcess = null;
+    logger.log("asking politely the internal process to close ...");
+    // internalProcess.removeListener("exit", handleExit);
+    internalProcess.disconnect();
+    setTimeout(() => {
+      if (internalProcess) {
```

Comment:
as a `killInternalProcess` can happen in many circumstances and concurrently (like switching some experimental settings) nothing seems to prevent here a race condition. I think what should be done is to make a local variable in the function scope so the inside of setTimeout don't use the global one, but the local one.